front/src/components/boxs/todolist/TodolistBox.jsx

```diff
@@ -0,0 +1,158 @@
+          </span>
+        </div>
+        {task.due && (
+          <div class="row">
+            <span style={{ opacity: '50%', fontSize: '0.80rem' }}>
+              <i className={`fe fe-calendar`} style={{ marginLeft: '20px', marginRight: '6px' }} />
+              <span>{dayjs(task.due.date).format('D MMM')}</span>
+            </span>
+          </div>
+        )}
+      </div>
+    </div>
+  );
+};
+
+const TodolistBox = ({ children, ...props }) => {
+  const onTaskClick = task => () => {
+    props.onTaskClick(task);
+  };
+  return (
+    <div class="card">
+      <div className="card-header">
+        <h3 class="card-title">
+          <i class="fe fe-list" />
+          <span class="m-1">{props.name || <Text id="dashboard.boxTitle.todolist" />}</span>
+        </h3>
+      </div>
+      <div>
+        {props.boxStatus === GetTodolistStatus.ServiceNotConfigured && (
+          <div class="card-body">
+            <p class="alert alert-danger">
+              <i class="fe fe-bell" />
+              <span class="pl-2">
+                <Text id="dashboard.boxes.todolist.serviceNotConfigured" />
+              </span>
+            </p>
+          </div>
+        )}
+        {props.boxStatus === RequestStatus.Error && (
+          <div class="card-body">
+            <p class="alert alert-danger">
+              <i class="fe fe-bell" />
+              <span class="pl-2">
+                <Text id="dashboard.boxes.todolist.unknownError" />
+              </span>
+            </p>
+          </div>
+        )}
+        {props.boxStatus === RequestStatus.Getting && !props.tasks && (
+          <div>
+            <div class="card-body">
+              <div class="dimmer active">
+                <div class="loader" />
+                <div class="dimmer-content" style={padding} />
+              </div>
+            </div>
+          </div>
+        )}
+        {props.tasks && (
+          <div style={padding} class="card-block o-auto">
+            <div>
+              {props.tasks.map(task => (
+                <Task key={task.id} task={task} onClick={onTaskClick(task)} />
+              ))}
+              {!props.tasks.length && (
+                <i>
+                  <Text id="dashboard.boxes.todolist.emptyTasks" />
+                </i>
+              )}
+            </div>
+          </div>
+        )}
+      </div>
+    </div>
+  );
+};
+
+@connect('DashboardBoxDataTodolist,DashboardBoxStatusTodolist', actions)
+class TodolistBoxComponent extends Component {
+  closeTaskUnbound(task) {
+    this.props.closeTask(task.id, this.props.x, this.props.y);
+  }
+
+  constructor(props) {
+    super(props);
+    this.closeTask = this.closeTaskUnbound.bind(this);
+  }
+
+  componentDidMount() {
+    const { todolist_id: todolistId } = this.props.box;
+
+    // get tasks
+    this.props.getTasks(todolistId, this.props.x, this.props.y);
+
+    // refresh tasks every interval
+    this.refreshInterval = setInterval(
```

Comment:
Don't we want to use events over web socket instead of interval refresh time?

Reply:
I cannot use websocket because the todoist api is on on RestApi. So I need to call the api to refresh the list of the task like the weather boxes (I don't like `setInterval` anymore but I don't find a best way to do this)

Reply:
What I mean is that the refresh interval can be done on backend side, so the server fetch the API every N seconds and notify the front end with a websocket event.

Reply:
Yes, It's a good idea. I try to do this :) Ty

Reply:
@atrovato : I have made the change, it's ok for you ?

Reply:
I don't necessarily agree on this change, because right now this integration doesn't store any data on Gladys side, so it means the backend will fetch Todoist API every minute even if the user is not on the frontend.

**We have 2 types of integration on Gladys:**

- The weather API is not storing any data, and we fetch it every 30 minutes **from the frontend**, so the API is only queried when necessary
- The calendar API stores data in Gladys local DB (for scenes for example), and refresh this every 30 minutes **backend side** so the DB is always up to date

This integration is very nice but it lacks of general reflexion on what it's trying to achieve. Is it more like the weather API or the calendar API ?